receivers/pagerduty/pagerduty.go

```diff
@@ -74,15 +79,28 @@ func (pn *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error
 		return false, fmt.Errorf("build pagerduty message: %w", err)
 	}
 
-	body, err := json.Marshal(msg)
-	if err != nil {
-		return false, fmt.Errorf("marshal json: %w", err)
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(msg); err != nil {
+		return false, fmt.Errorf("failed to encode PagerDuty message: %w", err)
+	}
+
+	if buf.Len() > pagerDutyMaxEventSize {
```

Comment:
let's leave a comment with reference to alertmanager code base to reason about this solution